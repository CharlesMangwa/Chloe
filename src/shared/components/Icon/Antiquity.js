/* @flow */
/* eslint max-len: ["error", 1000] */

import React from 'react'
import Svg, { Path } from 'react-native-svg'

type Props = {
  color: string,
  size: number,
}

const Antiquity = (props: Props) => {
  const { color, size } = props
  const ratio = 17 / 16
  return (
    <Svg width={ratio * size} height={size} viewBox="0 0 17 16">
      <Path
        d="M8.32.02L0 12l8.32 3.98 8.32-4"
        fill={color}
        fillRule="evenodd"
      />
    </Svg>
  )
}

export default Antiquity
