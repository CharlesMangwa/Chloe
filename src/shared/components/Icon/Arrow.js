/* @flow */

import React from 'react'
import { Svg, Path } from 'react-native-svg'

type Props = {
  color: String,
  size: Number,
}

const Arrow = (props: Props) => {
  const { color, size } = props
  const ratio = 8 / 14
  return (
    <Svg width={ratio * size} height={size} viewBox="0 0 8 14">
      <Path
        d="M0 1l8 6.116-8 6.116"
        fill="none"
        stroke={color}
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </Svg>
  )
}

export default Arrow
