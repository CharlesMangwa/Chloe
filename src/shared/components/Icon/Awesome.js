/* @flow */
/* eslint max-len: ["error", 1000] */

import React from 'react'
import Svg, { Ellipse, G, Path } from 'react-native-svg'

type Props = { size: number }

const Awesome = (props: Props) => {
  const { size } = props
  const ratio = 90 / 92
  return (
    <Svg width={ratio * size} height={size} viewBox="0 0 90 92">
      <G fill="none" fillRule="evenodd">
        <Ellipse fill="#FCDB74" cx="47.5" cy="49" rx="42.5" ry="42" />
        <Path d="M8.384 61.735c-.383 0-.775.025-1.17.063-1.254-4.005-1.93-8.263-1.93-12.68 0-23.395 18.963-42.36 42.357-42.36.49 0 .974.02 1.458.037-.28 2.156.004 4.41.816 6.633-.752-.048-1.51-.08-2.273-.08-19.723 0-35.768 16.046-35.768 35.77 0 5.363 1.193 10.45 3.317 15.02-1.927-1.545-4.26-2.403-6.806-2.403M80.41 22.28c-1.803 1.696-3.557 3.06-5.153 4.128 5.092 6.18 8.154 14.095 8.154 22.71 0 19.723-16.046 35.77-35.77 35.77-5.214 0-10.167-1.132-14.64-3.148-.622.973-1.35 1.924-2.177 2.84-.77.85-1.537 1.612-2.298 2.334 5.744 2.91 12.235 4.563 19.116 4.563C71.036 91.477 90 72.51 90 49.117c0-10.186-3.598-19.53-9.59-26.837" fill="#FCDB74" />
        <Path d="M40.424 46.46c.863 0 1.724-.337 2.37-1.007 1.265-1.31 1.228-3.395-.08-4.66-7.325-7.067-16.683-3.25-20.608-.244-1.44 1.102-1.712 3.155-.617 4.6 1.093 1.444 3.156 1.728 4.61.64 6.727-5.045 11.513-.76 12.037-.254.638.617 1.464.924 2.287.924M52.078 40.55c-1.44 1.102-1.712 3.155-.617 4.6 1.095 1.444 3.157 1.728 4.61.64 6.73-5.045 11.515-.76 12.038-.254.64.617 1.464.924 2.29.924.86 0 1.723-.337 2.37-1.007 1.262-1.31 1.226-3.395-.084-4.66-7.322-7.067-16.682-3.25-20.606-.244M58.104 57.253c-9.53 8.26-17.94.957-18.874.09-1.323-1.243-3.4-1.184-4.648.133-1.25 1.32-1.196 3.406.124 4.657 2.716 2.576 7.914 5.594 14.066 5.594 4.26 0 8.98-1.45 13.65-5.494 1.374-1.193 1.52-3.272.33-4.648-1.192-1.375-3.273-1.52-4.648-.332" fill="#000" />
        <Path d="M25.557 64.847c-.444-.06-.87-.088-1.278-.088-5.035 0-7.26 4.26-7.95 5.96-.686-1.7-2.914-5.96-7.946-5.96-.41 0-.834.028-1.28.087-7.308.978-10.06 9.928-3.023 17.703 3.987 4.403 7.94 6.875 10.62 8.192.616.303 1.165.544 1.63.735.468-.19 1.015-.432 1.632-.735 2.68-1.317 6.634-3.79 10.62-8.192 7.036-7.775 4.284-16.725-3.025-17.703M66.718 25.982c.617.304 1.165.545 1.63.735.468-.19 1.017-.43 1.633-.735 2.682-1.316 6.635-3.79 10.62-8.19 7.037-7.777 4.285-16.727-3.024-17.704-.444-.06-.87-.088-1.28-.088-5.032 0-7.258 4.26-7.947 5.96C67.66 4.26 65.434 0 60.4 0c-.408 0-.833.027-1.28.088-7.307.977-10.06 9.927-3.022 17.703 3.985 4.403 7.938 6.876 10.618 8.192" fill="#F15169" />
      </G>
    </Svg>
  )
}

export default Awesome
