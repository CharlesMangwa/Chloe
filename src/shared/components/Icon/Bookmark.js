/* @flow */

import React from 'react'
import { Svg, Path } from 'react-native-svg'

type Props = {
  color: String,
  size: Number,
}

const Arrow = (props: Props) => {
  const { color, size } = props
  const ratio = 11 / 15
  return (
    <Svg width={size * ratio} height={size} viewBox="0 0 11 15">
      <Path
        d="M9.87 1.19l-.066 12.532L5.5 10.05 1.196 13.72V1.192z"
        strokeWidth={1.5}
        stroke={color}
        fill={color}
        fillRule="evenodd"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </Svg>
  )
}

export default Arrow
