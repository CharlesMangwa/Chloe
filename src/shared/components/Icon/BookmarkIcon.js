/* @flow */

import React from 'react'
import Svg, { G, Path } from 'react-native-svg'

type Props = {
  color: string,
  size: number,
}

const BookmarkIcon = (props: Props) => {
  const { color, size } = props
  const ratio = 80 / 73
  return (
    <Svg width={ratio * size} height={size} viewBox="10 13 80 73">
      <G fill="none" fillRule="evenodd">
        <Path d="M88.26 17.4l-45.54-3.98c-.3-.03-.63.07-.87.27s-.4.5-.42.8l-.25 2.9H25.03c-.66 0-1.2.5-1.2 1.2v5.7l-12.33 1c-.32 0-.6.16-.8.4-.22.23-.3.54-.3.86l5.04 57.5c.06.62.58 1.1 1.2 1.1h.1l45.53-4c.32 0 .6-.18.8-.4.22-.26.3-.57.3-.9l-.23-2.46h7.6c.66 0 1.2-.52 1.2-1.2l11.08 1h.1c.3 0 .56-.1.77-.28.2-.2.4-.5.4-.82l5-57.5c.02-.63-.4-1.2-1.1-1.27z" />
        <Path d="M38.88 71l4.82-55.1 43.16 3.77-4.8 55.1" fill={color} />
        <Path d="M36.4 71.98c-.06.66.43 1.23 1.1 1.3l20.8 1.82H26.2V19.78h14.75l-4.57 52.2z" fill={color} />
        <Path d="M60.87 78.97L17.7 82.75l-4.8-55.1 10.94-.97v49.6c0 .67.53 1.2 1.2 1.2h35.7l.13 1.5z" fill={color} />
      </G>
    </Svg>
  )
}

export default BookmarkIcon
