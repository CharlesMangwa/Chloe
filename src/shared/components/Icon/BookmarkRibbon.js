/* @flow */

import React from 'react'
import Svg, { Path } from 'react-native-svg'

type Props = {
  color: string,
  size: number,
}

const BookMarkRibbon = (props: Props) => {
  const { color, size } = props
  const ratio = 20 / 33
  return (
    <Svg width={ratio * size} height={size} viewBox="162 0 20 33">
      <Path
        d="M162 0h20v30.97c0 1.1-.72 1.47-1.6.82l-8.4-6.17-8.4 6.15c-.88.64-1.6.27-1.6-.83V0z"
        fill={color}
        fillRule="evenodd"
      />
    </Svg>
  )
}

export default BookMarkRibbon
