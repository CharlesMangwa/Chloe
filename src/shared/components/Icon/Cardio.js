/* @flow */
/* eslint max-len: ["error", 2000] */

import React from 'react'
import Svg, { G, Path } from 'react-native-svg'

type Props = {
  color: string,
  size: number,
}

const Cardio = (props: Props) => {
  const { color, size } = props
  const ratio = 44 / 39
  return (
    <Svg width={ratio * size} height={size} viewBox="0 1 44 39">
      <G fill="none" fillRule="evenodd">
        <Path
          d="M6.593 22.278c.11-.256.187-.536.338-.765.4-.6.99-.83 1.7-.825 2.15.013 4.29.01 6.44.022.21 0 .36-.054.5-.22 3.21-3.738 6.43-7.47 9.65-11.204l.23-.273c-.12-.008-.2-.02-.28-.02-.69-.003-1.38.002-2.062-.013-.207-.004-.34.063-.47.216-1.424 1.656-2.857 3.303-4.28 4.96-.417.485-.91.748-1.56.612-1.066-.222-1.537-1.53-.827-2.356 1.68-1.957 3.38-3.898 5.076-5.846.335-.383.756-.56 1.26-.56 2.31 0 4.62-.002 6.93-.006.636 0 1.16.235 1.607.69 1.39 1.4 2.786 2.788 4.18 4.18.07.075.15.146.24.23.09-.083.164-.154.24-.228 1.11-1.11 2.254-2.193 3.33-3.34.92-.978 2.426-.538 2.67.64.007.027.03.05.048.073v.524c-.134.48-.464.83-.81 1.17-1.32 1.31-2.63 2.64-3.942 3.96-.162.16-.333.32-.52.46-.742.53-1.54.47-2.186-.17-.834-.83-1.66-1.67-2.49-2.5-.072-.08-.15-.15-.25-.24L26.99 16.5c.08.09.15.16.223.24 1.205 1.21 2.406 2.42 3.624 3.62.6.59.76 1.3.59 2.08-.757 3.433-1.53 6.862-2.3 10.294-.317 1.396-1.67 2.04-2.85 1.36-.69-.395-1.066-1.224-.89-2.02.59-2.667 1.19-5.33 1.795-7.996.05-.213.02-.35-.144-.51-1.4-1.345-2.788-2.7-4.18-4.053l-.208-.183c-.05.075-.087.16-.14.22-1.2 1.34-2.406 2.667-3.59 4.014-.525.59-1.196.84-1.94.85-2.806.03-5.61.04-8.415.04-1.033-.006-1.69-.58-1.93-1.596-.005-.025-.03-.05-.04-.07v-.528z"
          fill={color}
        />
        <Path
          d="M32.67 1c.63.175 1.234.39 1.75.827 1.42 1.204 1.45 3.59.033 4.804-.986.85-2.11 1.11-3.33.65-1.21-.46-1.9-1.38-2.092-2.66-.23-1.61.93-3.17 2.63-3.54.06-.01.12-.04.18-.06h.84z"
          fill={color}
        />
        <Path
          d="M5 8.208h9.25m-13 5H12.5m-9.25 4.25H14.5M2.25 27.333H13.5m-8.25 4.25H16.5"
          stroke={color}
          fill={color}
        />
      </G>
    </Svg>
  )
}

export default Cardio
