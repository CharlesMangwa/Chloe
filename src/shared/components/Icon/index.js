/* @flow */
/* eslint react/jsx-indent: 0 */

import React, { createElement } from 'react'
import { View } from 'react-native'
import { rem } from '@helpers/responsive'
import { BRAND_COLOR_50, NEUTRAL_COLOR_55 } from '@theme/colors'

type Props = {
  name: string,
  actived?: boolean,
  size?: number,
  defaultActivedColor?: string,
  defaultColor?: string,
  style?: any,
}

const Icon = (props: Props): React$Element<any> | null => {
  const { name, style, size, actived, defaultColor, defaultActivedColor } = props
  const color = actived
    ? defaultActivedColor || BRAND_COLOR_50
    : defaultColor || NEUTRAL_COLOR_55
  let icon
  switch (name) {
    case 'arrow':
      icon = require('./Arrow').default
      break
    case 'awesome':
      icon = require('./Awesome').default
      break
    case 'bookmark':
      icon = require('./Bookmark').default
      break
    case 'cardio':
      icon = require('./Cardio').default
      break
    case 'chill':
      icon = require('./Chill').default
      break
    case 'close':
      icon = require('./Close').default
      break
    case 'dead':
      icon = require('./Dead').default
      break
    case 'fail':
      icon = require('./Fail').default
      break
    case 'filter':
      icon = require('./Filter').default
      break
    case 'historic':
      icon = require('./Historic').default
      break
    case 'hit':
      icon = require('./Hit').default
      break
    case 'like':
      icon = require('./Like').default
      break
    case 'location':
      icon = require('./Location').default
      break
    case 'muscle':
      icon = require('./Muscle').default
      break
    case 'navigation':
      icon = require('./Navigation').default
      break
    case 'phone':
      icon = require('./Phone').default
      break
    case 'search':
      icon = require('./Search').default
      break
    case 'share':
      icon = require('./Share').default
      break
    case 'star':
      icon = require('./Star').default
      break
    case 'stretching':
      icon = require('./Stretching').default
      break
    case 'success':
      icon = require('./Success').default
      break
    case 'marker':
      icon = require('./Marker').default
      break
    default:
  }
  return icon
    ? <View style={style}>
        {createElement(
          icon,
          { color, size }
        )}
      </View>
    : null
}

Icon.defaultProps = {
  actived: false,
  size: rem(.875),
}

export default Icon
