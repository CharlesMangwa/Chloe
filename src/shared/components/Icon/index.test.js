import 'react-native'
import renderer from 'react-test-renderer'
import React from 'react'
import Icon from '@components/Icon'
jest.mock('react-native-svg', () => 'Icon')

describe('Shared component: <Icon />', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <Icon />
    ).toJSON()
    expect(component).toMatchSnapshot()
  })
})
