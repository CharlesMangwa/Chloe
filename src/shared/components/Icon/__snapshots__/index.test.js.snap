exports[`Shared component: <Icon /> renders correctly 1`] = `null`;
